### ISSUE TITLE
[Suggestion] Fall damage numeric estimation

### ISSUE THREAD

**Issue by budak7273** [Basement-Science/SatisfactoryMod_Speedo#2]
You might be able to take the fall damage estimation color feature a step further by plugging in the player's velocity into the fall damage curve. I'm not sure exactly where it is, but if you [generate a full project dump](https://docs.ficsit.app/satisfactory-modding/latest/CommunityResources/AssetToolkit.html) and filter in the editor by data curve assets, you'll probably find it.

**Comment by Basement-Science** [comment]
yeah I may look into that. I thought about changing the icon to the skull when the damage will actually kill you based on your health, but didn't find a quick solution at the time. But then I dont know much about unreal and modding, and hadn't looked into this Asset Toolkit at all either. 

In general I wanted to get a few status attributes about the player that I couldn't find yet, like being in a vehicle for example.

**Comment by budak7273** [comment]
If you do pursue this, here's the paths for the curves:

Normal
`CurveFloat'/Game/FactoryGame/Character/Player/FallDamageCurve.FallDamageCurve'`
![image](https://user-images.githubusercontent.com/25965766/151742296-7043bcac-d93e-45d3-8ec9-d6687ce771df.png)

Blade Runners
`CurveFloat'/Game/FactoryGame/Equipment/JumpingStilts/JumpingStilts_FallDamageCurve.JumpingStilts_FallDamageCurve'`
![image](https://user-images.githubusercontent.com/25965766/151742349-5b250a12-9ab0-49f0-95c7-0cf18681ca38.png)

I'm not sure offhand how to detect being in a vehicle, but I think the Pawn possessed by the player controller will change when this is the case. Pawns/Controllers are an Unreal concept. You can get the local player controller via the blueprint node [GetPlayerController](https://docs.unrealengine.com/4.26/en-US/BlueprintAPI/Game/GetPlayerController/) or you can get it from an actor's instigator.

**Comment by Basement-Science** [comment]
I tried getting the dump tool working but alpakit failed to build it for whatever reason. Didn't find a solution for that. 
It just complained about not finding a file it was working on itself. No idea what's up with that. Maybe some future version will fix it.

**Comment by MinoDab492** [comment]
> Owner

You need to compile the project in VS before you alpakit. If you didn't do that already.

**Comment by Basement-Science** [comment]
> You need to compile the project in VS before you alpakit. If you didn't do that already.

I found the solution in the meantime. Alpakit was upset at the directory structure I used even though unreal editor is fine with it.

**Comment by MinoDab492** [comment]
> > You need to compile the project in VS before you alpakit. If you didn't do that already.
> 
> I found the solution in the meantime. Alpakit was upset at the directory structure I used even though unreal editor is fine with it.

Great to hear! I was looking for ideas from the SF Modding Discord Server, and I was given the idea to make a speedometer for Satisfactory in MPH. I was wondering if you would be willing to teach me the basics of how the mod hooks in, and get's the speed, and all of that stuff?

**Comment by budak7273** [comment]
@MinoDab492 The mod is open source, you should be able to clone it into a plugin folder in your current Unreal project and explore how the mod is set up from there

**Comment by MinoDab492** [comment]
True, but I more meant could they help me actually understand what each thing does and all that.

**Comment by Basement-Science** [comment]
> True, but I more meant could they help me actually understand what each thing does and all that.

So did you see that I opened another thread for you? Or have you lost interest?
